tests/test-highlighting.zsh

```diff
@@ -137,10 +142,10 @@ run_test_internal() {
     local desc="[$start,$end] «${BUFFER[$start,$end]//'#'/♯}»"
     # Match the emptiness of observed_result if no highlighting is expected
     [[ $highlight_zone[3] == NONE ]] && highlight_zone[3]=
-    [[ -n "$highlight_zone[4]" ]] && todo="# TODO $highlight_zone[4]"
+    (( $+highlight_zone[4] )) && todo="# TODO $highlight_zone[4]"
     for j in {$start..$end}; do
-      if [[ "$observed_result[$j]" != "$highlight_zone[3]" ]]; then
-        print -r -- "not ok $i - $desc - expected ${(qqq)highlight_zone[3]}, observed ${(qqq)observed_result[$j]}. $todo"
+      if [[ "${observed_result[$j]-}" != "$highlight_zone[3]" ]]; then
```

Comment:
I wonder if instead of ${foo-} here, we should explicitly test for undefinedness and pair it with the magic NONE value:

```zsh
if if (( ${+observed_result[$j]} )); then [[ $highlight_zone[3] == NONE ]]; else [[ $highlight_zone[3] == ${observed_result[$j] ]]; fi; then
```

(Well, rather, something equivalent to this but _slightly_ more readable ☺...)